align_anything/evaluation/benchmarks/mt_bench/vllm_eval.py

```diff
@@ -0,0 +1,342 @@
+            score = get_score(output)
+            time = 0
+            while score is None:
+                multi_results=[]
+                multi_results = judger.evaluate(system_prompts=[system_prompt],user_prompts=[user_prompt])
+                output =multi_results[0].raw_output.choices[0].message.content
+                score = get_score(output)
+                time+=1
+                if time >=10:
+                    score = 0
+                    break
+            
+            eval_case.append(
+                 {
+                    'question_id' : response['question_id'],
+                    'system_prompt': system_prompt,
+                    'user_prompt' : user_prompt,
+                    'response': output,
+                    'score': score
+                 }
+             )       
+    
+    return responses, eval_case
+
+def main():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    _, unparsed_args = parser.parse_known_args()
+    keys = [k[2:] for k in unparsed_args[0::2]]
+    values = list(unparsed_args[1::2])
+    unparsed_args = dict(zip(keys, values))
+    logger = EvalLogger('Evaluation')
+    
+    dict_configs, infer_configs = read_eval_cfgs('mt_bench', 'vLLM')
+
+    try:
+        assert dict_configs or infer_configs, "Config file does not exist or is incomplete."
+    except AssertionError as e:
+        logger.log('error', "Config file is not exist or incomplete.")
+        exit()
+
+    for k, v in unparsed_args.items():
+        dict_configs = update_dict(dict_configs, custom_cfgs_to_dict(k, v))
+        infer_configs = update_dict(infer_configs, custom_cfgs_to_dict(k, v))
+
+    dict_configs, infer_configs = dict_to_namedtuple(dict_configs), dict_to_namedtuple(infer_configs)
+    
+    model_config = dict_configs.default.model_cfgs
+    eval_configs = dict_configs.default.eval_cfgs
+
+    dataloader = MTBenchDataLoader(dict_configs)
+    assert not (dataloader.num_shot > 0 and dataloader.cot), "Few-shot and chain-of-thought cannot be used simultaneously for this benchmark."
+    test_data_round1 = dataloader.load_dataset()
+    eval_module = MTBenchGeneratorVLLM(model_config, infer_configs)
+    output_data_round1 = eval_module.eval(test_data_round1, eval_configs)
+    
+    test_data_round2 = dataloader.load_dataset_round2(output_data_round1)
+    output_data_round2 = eval_module.eval(test_data_round2, eval_configs)
+    file_name = f'{eval_module.model_id}'+f'_cot_{dict_configs.default.eval_cfgs.cot}'
+    responses=[]
+    evals=[]
+    for task, _ in output_data_round2.items():
+        responses,evals = evaluator(output_data_round1[task],output_data_round2[task], dataloader, task, eval_configs)
+    
+    merged_list=[]
+    for resp, eval_ in zip(responses, evals):
+        merged_dict = {**resp, **eval_}
+        merged_list.append(merged_dict)
+
+    raw_result_file = eval_configs.output_dir+file_name+"_raw_result.jsonl"
```

Comment:
```suggestion
    os.makedirs(eval_configs.output_dir,exist_ok=True)
    raw_result_file = os.path.join(eval_configs.output_dir,file_name + "_raw_result.jsonl")
```

Bad output file path

Reply:
Thanks for your insights and suggestions!